### PR TITLE
docs: Remove refs for distributed caching

### PR DESCRIPTION
**Pull request by vikram-dagger**

### CODE REVIEW

**Comment by vikram-dagger** [comment]
@marcosnils @kpenfound can you please review the revised snippets esp. the removal of `docker stop`?

@gerhard can you please review the changes to the Kubernetes page?

File: docs/current_docs/integrations/kubernetes.mdx [hunk 1]
```diff
@@ -67,7 +67,7 @@ The previous pattern makes it possible to scale the Dagger deployment, but comes
 
 In this architecture:
 
-- A shared cloud cache stores data from all Dagger Engines running in the cluster. This diagram below references Dagger's experimental Cloud Cache service.
+- A shared cloud cache stores data from all Dagger Engines running in the cluster.
```
Comment (kpenfound) [inline docs/current_docs/integrations/kubernetes.mdx:70]: I see the diagram is updated but the whole "cloud cache" circle should be removed

File: docs/current_docs/configuration/cloud.mdx [hunk 2]
```diff
@@ -227,7 +218,7 @@ You cannot change a member's role at this time. Please contact Dagger via the su
 ## Cache volumes
 
 :::warning
-Dagger Cloud's distributed caching feature is currently experimental.
+Dagger Cloud's caching feature is currently experimental.
```
Comment (kpenfound) [inline docs/current_docs/configuration/cloud.mdx:221]: This whole section needs to be removed

File: docs/current_docs/integrations/kubernetes.mdx (hunk 1 above)
Comment (vikram-dagger) [reply docs/current_docs/integrations/kubernetes.mdx:70]: Is this pattern invalid without Dagger Cloud? ie. a user cannot create and drop in their own homebrew cloud cache solution?

File: docs/current_docs/configuration/cloud.mdx (hunk 2 above)
Comment (vikram-dagger) [reply docs/current_docs/configuration/cloud.mdx]: Done

**Review by gerhard — APPROVED** [review]
I added a few atomic commits.

Feel free to change / drop and commit when ready @vikram-dagger

**Pull request merged**